projects/inchi/inchi_input_fuzzer.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <limits>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+
+#include "inchi_api.h"
+
+static constexpr size_t kSizeMax = std::numeric_limits<size_t>::max();
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+
+  if (size == kSizeMax)
+    return 0;
```

Comment:
Do you perform this check because you have to add the \0 terminator?  (Is it really a problem if the buffer that's used as input for `GetINCHIKeyFromINCHI` is `size_t + 1` bytes long?)

Reply:
Yes, I just didn't want problems to arise if the size of the fuzzing input is the maximum value for size_t and `szINCHISource` then gets initialized to zero bytes.